Azure Monitor: Blank result formats are not overwritten

bossinc opened the pull request:
moved to https://github.com/grafana/grafana/pull/93493

bossinc committed:
if result format is null from previous query, it will be null and not correct to a different format

bossinc committed:
if result format is null from previous query, it will be null and not correct to a different format

aangelisc reviewed (commented):
This logic was handled in the `ParseResultFormat` function already. I think the behaviour we want is to set the `ResultFormat` to `ResultFormatTimeSeries` to match the previous functionality.

Similarly for the frontend, if there are no queries defined then I think the logic you've added is okay. However, if there is a query defined but it doesn't have a result format (and it's a logs query) again that should default to `time_series` rather than `logs`.

bossinc committed:
update to not have empty value in result format field

bossinc committed:
update to not have empty value in result format field

bossinc committed:
gfg

bossinc committed:
fix tests

bossinc committed:
fix tests

gamab committed:
RBAC: AccessControlOnCall general availability (#93383)

RBAC: GA accessControlOnCall

academo committed:
Chore: Fix issue-labeled workflow permission errors (#93417)

* Chore: Fix issue-labeled workflow permission errors

* fix comparison

kaydelaney committed:
Dashboards: Fix 'days' unit pluralization (#93326)

alexanderzobnin committed:
Docs: Add announcement banners to list of RBAC actions (#93421)

lucamotion committed:
Dashboards: Fix typo in PublicDashboardNotAvailable component (#93418)

juanicabanas committed:
ShareDrawer: Update documentation link buttons (#93339)

eleijonmarck committed:
Docs: Add http docs for lbac for datasources (#93399)

* Add: http docs for lbac for datasources

* spelling

* update with only cloud loki

* rename to lbac for datasources

* moved it

* Update _index.md

This commit fixes minor style and punctuation issus

* change datasource to data source

* replace datasource with data source

minor updates and style fixes

* minor style changes

* prettier

---------

Co-authored-by: Irene Rodriguez <irene.rodriguez@grafana.com>

aangelisc reviewed (commented):
I think tests are failing now because the previous logic that would set the value to the default if the user has specified a value that isn't supported has been removed

JohnnyQQQQ committed:
fix(apiserver): append errors correctly from options (#93434)

aangelisc committed:
AzureMonitor: Improve resource picker efficiency (#93127)

* Parameterise region building metric namespace URL

- Add parameter for region (this parameter takes precedence over if global is set)
- Update tests
- Support this parameter on the data source method

* Refactor fetchAllNamespaces

- Use Set rather than an array for greater performance
- Request namespaces across WestEurope, EastUS, and JapanEast concurrently
- Update test

* Maintain existing behaviour

itsmylife committed:
Chore: Fix running frontend unit tests in JetBrains (#93441)

read the files with their absolute paths

torkelo committed:
DashboardScene: Fix url state sync issue when coming form Home dashboard to a cached dashboard scene (#93442)

* DashboardScene: Fixes preserve dashboard state for hidden options

* DashboardScene: Alt fix for pervering only specific url keys

* DashboardScene: Fixes url sync issue when coming from Home route

* Update

* Fix test

bossinc committed:
add check to see if options are in list

bossinc committed:
add check to see if options are in list

JohnnyQQQQ committed:
feat(apiserver): add flag to control the request size limit (#93431)

bossinc committed:
reformat stuff so logic lives on the right layer

bossinc committed:
reformat stuff so logic lives on the right layer

bossinc committed:
linterrrrrrrrrrrrrrrrrrRR

bossinc committed:
linterrrrrrrrrrrrrrrrrrRR

masterwishx committed:
Chore: Add ILS (Israeli New Shekels) currency to value formatter (#93390)

add ILS to Currency

ashharrison90 committed:
Select: Fix last option in an option group to be selectable (#93457)

don't completely overwrite innerProps

ryantxu committed:
Plugins: Expose datasources as apiservers feature toggle (#93250)

academo committed:
Chore: Change condition to run auto triager based on user association (#93466)

ryantxu committed:
K8s/Dashboard: Read DTO from the registered storage, not just legacy (#93406)

gamab committed:
AuthN: Ext JWT support actions (#92486)

jdbaldry committed:
Update workflows to use actions that don't need organization secrets (#92603)

wbrowne committed:
Plugins: Avoid returning 404 for `AutoEnabled` apps (#93436)

avoid errors for autoenabled apps

bossinc commented on pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go:
```diff
@@ -172,6 +172,10 @@ func buildLogAnalyticsQuery(query backend.DataQuery, dsInfo types.DatasourceInfo
 	resultFormat := ParseResultFormat(azureLogAnalyticsTarget.ResultFormat, dataquery.AzureQueryTypeAzureLogAnalytics)
+	if azureLogAnalyticsTarget.ResultFormat != nil {
+		resultFormat = *azureLogAnalyticsTarget.ResultFormat
+
```
```suggestion
```

bossinc committed:
Apply suggestions from code review

linoman committed:
SSO LDAP: Fix ldap provider in providers list (#93472)

* Correct comment about provider filter

* Add type for LDAP provider

* Ignore old LDAP init setup when using SSO

ryantxu committed:
ResourceClient: Exercise resource client in k8s apis tests (#93473)

bohandley committed:
Explore Metrics: Get OTel resources and filter metrics and labels (#91221)

* add OTel filter in metric select scene

* add resource query to get matching OTEL job&instance

* filter metrics by OTEL resources

* only add otel select if DS has OTEL matching job and instance

* add folder for otel resources

* upate metric select for new otel folder

* move otel api call

* get otel resources for labels for single series job/instance target_info

* add otel resources to adhoc variable dropdown

* update otel api to check for standardization and return labels

* label types for api

* check standardization, show otel variable, select depenv, update other variables

* remove otel target list from metric select scene

* load resources if dep_env label has already been selected

* exclude previously used filters

* do not check standardization if there are already otel filters

* drop filters when switching data sources

* add experience var for switching to otel experience

* remove otel from variables and place near settings

* add error for non-standard prom with otel resources

* fix typescript errors, remove ts-ignores

* add custom variable for deployment environment like app-olly

* fix name of otel variable

* add function for getting otel resources from variables

* add otel join query const

* update standard check to be simpler

* allow for unstandard otel data sources but give warning

* add otelJoinQuery to the base query and clean up variables when state changes

* refactor otel functions to return filters for targets, use targets to filter metrics

* update metric names on otel target filter change

* when no otel targets for otel resource filter, show no metrics

* move switch to settings, default to use experience, refactor otel checks

* clean code

* fix refactor to add hasOtelResources for showing the switch in settings

* sort otel resources by blessed list

* reset otel when data source is changed

* move otel experience toggle back outside settings

* move showPreviews into settings

* do not re-add otel resources from blessed list to filters when already selected

* add otel join query variable to histogram base query

* only show settings for appropriate scenes

* show info tooltip the same but show error on hover for disabling otel exp for unstandard DS

* refactor tagKeys and tagValues for otel resources variable, fix promoted list ordering, fix dep env state bug

* default dep env value

* apply var filters only where they are using VAR_FILTER_EXPR in queryies

* change copy for labels to attributes

* do not group_left job label when already joining by job

* update copy for label variable when using otel

* remove isStandard check for now because of data staleness in Prometheus

* default to showing heatmap for histograms

* add trail history for selecting dep env and otel resources

* add otel resource attributes tests for DataTrail

* move otel functions to utils

* write tests for otel api calls

* write tests for otel utils functions

* fix history

* standard otel has target_info metric and deployment_environment resource attributes

* fix tests

* refactor otel functions for updating state and variables

* clean code

* fix tests

* fix tests

* mock checkDataSourceForOtelResources

* fix tests

* update query tests with otelJoinQuery and default to heatmap for _bucket metrics

* fix tests for otel api

* fix trail history test

* fix trail store tests for missing otel variables

* make i18n-extract

* handle target_info with inconsistent job and instance labels

* fix otel copy and <Trans> component

* fix custom variable deployment environment bug when switchiing data sources from non otel to otel

* fix linting error for trans component

* format i18nKey correctly

* clean up old comments

* add frontend hardening for OTel job and instance metric list filtering

* fix test for deployment environment custom variable to use changeValueTo

* fix i18n

* remove comments for fixed bug

* edit skipped tests

bossinc committed:
frontend linter

bossinc committed:
Merge remote-tracking branch 'refs/remotes/origin/andrew/azure-monitor-doesnt-set-result-format' into andrew/azure-monitor-doesnt-set-result-format
